src/database/models/image_item.rs

```diff
@@ -0,0 +1,275 @@
+use super::ids::*;
+use crate::{database::models::DatabaseError, models::images::ImageContext};
+use chrono::{DateTime, Utc};
+use redis::cmd;
+use serde::{Deserialize, Serialize};
+
+const IMAGES_NAMESPACE: &str = "images";
+const DEFAULT_EXPIRY: i64 = 1800; // 30 minutes
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct Image {
+    pub id: ImageId,
+    pub url: String,
+    pub size: u64,
+    pub created: DateTime<Utc>,
+    pub owner_id: UserId,
+
+    // context it is associated with
+    pub context: String,
```

Comment:
context doesn't need to exist IMO since you have project/version/thread message/report ids